packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+audit `My audit` and the resulting value `0`.
+
+## Plugin output
+
+Every plugin executes audits and returns the outcome as [`AuditOutputs`](@TODO), and array of [`AuditOutput`](@TODO).
+
+The minimum output of an audit looks like this:
+
+```typescript
+import { AuditOutput } from '@code-pushup/models';
+
+const myAuditOutput: AuditOutput = {
+  ...myAuditMeta,
+  score: 0,
+  value: 0,
+  // optional details
+};
+```
+
+- A audit output always includes the metadata of the audit. [`Audit`](@TODO)
+
+- `score` and `value` are important to calculate a score out of a given metrics and display it.
+  Here you can read more about [audits and scoring](@TODO - in page link).
+
+- `details` helps with attribution of audit results. This is important to get actionable feedback like the line of code or how to fix it.  
+  Here you can read more on [attribution of audits](@TODO - in page link).
+
+Here an example of using the above audit for the plugin output:
+
+```typescript
+import { AuditOutputs } from '@code-pushup/models';
+
+const pluginOutput: AuditOutputs = [myAuditOutput];
+```
+
+## Plugin runner
+
+The core of a plugin is defined under the `runner` property.
+The `runner` property is the entry point of your plugin and is called by the CLI. It should return the audit results
+as [`AuditOutputs`](@TODO).
+
+A plugins runner logic can get implemented in 2 ways:
+
+- as a `RunnerFunction`
+- as a `RunnerConfig`
+
+Even if both of them result in [`AuditOutputs`](@TODO), we recommend the `RunnerFunction` for getting started.
+It is easier to use for simple plugins and can be written in the config file directly.
+The `RunnerConfig` is suitable for more complex, performance-heavy plugins (runner executed off the main thread), and is more flexible in regard to runtime (can run any shell command, not restricted to JavaScript).
+
+### RunnerFunction
+
+The `RunnerFunction` is the entry point of your plugin. It is called by the CLI and should return the audit results
+as `AuditOutputs`.
+
+Let's write a real implementation to get familiar with the runner function.
+We will implement a simple file size audit for JavaScript files that tracks the size of specified files in your
+codebase.
+
+1. Use the template from the section [Plugin Structure](#Plugin-Structure) as a starting point and fill in the correct information for the plugin metadata.
+2. Add the `directory` property to the plugin options and use the plugin in you config file.
+
+```typescript
+// file-size.plugin.ts
+import { AuditOutput, AuditOutputs } from '@code-pushup/models';
+
+export type Options = {
+  // add the directory to the plugin options
+  directory: string;
+};
+
+const fileSizeAuditMeta: AuditOutput = {
+  slug: 'file-size',
+  title: 'File size',
+  // ...
+};
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  return [
+    {
+      ...fileSizeAuditMeta,
```

Comment:
The runner shouldn't produce all the metadata, just the `slug` to identify the audit:

```suggestion
      slug: fileSizeAuditMeta.slug,
```